### ISSUE TITLE
Koin eating up crash reports

### ISSUE THREAD

**Issue by abhishekBansal** [InsertKoinIO/koin#845]
**Describe the bug**
Koin is eating up actual crash reports as my app crashes twice most of the time an actual crash happens. For example lets say there is an `IndexOutOfBounds` exception in some activity and it crashes. Immediately after that previous activity crashes with no definition found exception and previous report is not visible in logcat or Crashlytics. Its making debugging production crashes impossible.

I load modules in two steps. One before and one after authentication in my app, also, currently I am not using any scopes. 

Koin version is `2.0.1`
Crash log looks like this no matter what the previous activity was.
```
Caused by k.a.c.f.e: No definition found for '<package>.h' has been found. Check your module definitions.
       at org.koin.core.scope.Scope.findDefinition(Scope.java:52)
       at org.koin.core.scope.Scope.resolveInstance(Scope.java)
       at org.koin.core.scope.Scope.get(Scope.java:123)
       at org.koin.androidx.viewmodel.ViewModelResolutionKt$createViewModelProvider$1.create(ViewModelResolutionKt.java:25)
       at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:37)
       at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:23)
       at org.koin.androidx.viewmodel.ViewModelResolutionKt.getInstance(ViewModelResolutionKt.java:119)
       at org.koin.androidx.viewmodel.ViewModelResolutionKt.getViewModel(ViewModelResolutionKt.java:26)
```

**Update**
This does not happen 100% of the time. In one more scenario after a proper crash when user presses `Open App Again` application again crashes with above `Koin` stacktrace. And weirdly when `Open App Again` is pressed this time App runs successfully.

**Comment by abhishekBansal** [comment]
I think I figured out what was the issue. In our current implementation we are there are two type of modules 
1. Authenticated which logged user can access
2. UnAuthenticated which are used for onboarding and such

In our `Application` class we load second type of modules. And first type of modules are loaded in home activity or `deeplink` entry points if and when user is logged in. 

Now after app crashes my guess is system tries to re-create previous activity (in some phones automatically in some after clicking on Open App Again) Since `Application` class only loads 2nd type of modules app crashes because there are no definitions available via `Koin`

I have tested and confirmed this but now, what is the best way to do this? 
Currently I have placed this in every `Activity` as a workaround
```kotlin
fun login() {
        try {
            loadKoinModules(listOf(authenticatedModules, authenticatedViewModels))
        } catch (e: DefinitionOverrideException) {
            // fail silently
        }
    }
```
